apis/core/v1alpha1/helper/helpers.go

```diff
@@ -32,24 +39,29 @@ func SetOperation(obj *metav1.ObjectMeta, op v1alpha1.Operation) {
 	metav1.SetMetaDataAnnotation(obj, v1alpha1.OperationAnnotation, string(op))
 }
 
-// HasReconcileTimestampAnnotation checks if the obj has the given timeout annotation
-func HasReconcileTimestampAnnotation(obj metav1.ObjectMeta) bool {
-	_, ok := obj.Annotations[v1alpha1.ReconcileTimestampAnnotation]
+// HasTimestampAnnotation checks if the obj has the given timeout annotation
+func HasTimestampAnnotation(obj metav1.ObjectMeta, ta TimestampAnnotation) bool {
+	_, ok := obj.Annotations[string(ta)]
 	return ok
 }
 
-func GetReconcileTimestampAnnotation(obj metav1.ObjectMeta) (time.Time, error) {
-	return time.Parse(time.RFC3339, obj.Annotations[v1alpha1.ReconcileTimestampAnnotation])
+func GetTimestampAnnotation(obj metav1.ObjectMeta, ta TimestampAnnotation) (time.Time, error) {
+	return time.Parse(time.RFC3339, obj.Annotations[string(ta)])
+}
+
+// SetTimestampAnnotationNow sets the timeout annotation with the current timestamp.
+func SetTimestampAnnotationNow(obj *metav1.ObjectMeta, ta TimestampAnnotation) {
+	SetTimestampAnnotation(obj, ta, time.Now())
 }
 
-// SetReconcileTimestampAnnotationNow sets the timeout annotation with the current timestamp.
-func SetReconcileTimestampAnnotationNow(obj *metav1.ObjectMeta) {
-	SetReconcileTimestampAnnotation(obj, time.Now())
+// SetTimestampAnnotation sets the timeout annotation with the given timestamp.
+func SetTimestampAnnotation(obj *metav1.ObjectMeta, ta TimestampAnnotation, ts time.Time) {
+	metav1.SetMetaDataAnnotation(obj, string(ta), ts.Format(time.RFC3339))
 }
```

Comment:
this function is only used in one package and it does abstract that much.
I would suggest to either remove the helper and just call directly `metav1.SetMetaDataAnnotation(obj, string(ta), ts.Format(time.RFC3339))` or move the function(s) to the respective package

Reply:
I removed that one and only kept the `SetTimestampAnnotationNow` method - is this ok or would you like me to move it out of this package too?